common/src/service/ExceptionHandlingHelper.cs

```diff
@@ -37,5 +38,21 @@ public static Task<string> GetExceptionMessageAsync(HttpResponseMessage response
             // TODO: pradeepc - consider parsing to HttpError
             return response.Content.ReadAsStringAsync();
         }
+
+        private static async Task<Exception> HttpNotFoundMapping(HttpResponseMessage response)
+        {
+            string exMsg = await GetExceptionMessageAsync(response).ConfigureAwait(false);
+            string httpErrorCode = response.Headers.GetFirstValueOrNull(CommonConstants.HttpErrorCodeName);
+            ErrorCode errorCode;
+            if (Enum.TryParse(httpErrorCode, out errorCode))
+            {
+                switch (errorCode)
+                {
+                    case ErrorCode.DeviceNotOnline:
+                        return new DeviceNotOnlineException(exMsg);
+                }
+            }
+            return new DeviceNotFoundException(exMsg);
+        }
```

Comment:
Should it return the generic exception DeviceNotFoundException in all other cases?

Reply:
Can you add a test for this scenario as well?

Reply:
It's been returning generic DeviceNotFoundException before, it's just according to exception in http headers. But DeviceNotFound and DeviceNotOnline both have 404 error code, that's why the issue appeared